packages/dashboard/src/components/sidePanel/sections/chartSettings.tsx

```diff
@@ -1,11 +1,124 @@
+      </div>
+    </SettingTile>
+  );
+};
+
+const SizeSettings = () => {
+  const [width, updateWidth] = useInput<number>('width');
+  const [height, updateHeight] = useInput<number>('height');
+  return (
+    <SettingTile title={'Size'} gridDefinition={[{ colspan: 6 }, { colspan: 6 }]}>
+      <div className="section-item-content">
+        <div className="section-item-label">Width</div>
+        <Input value={`${width}`} type="number" onChange={({ detail: { value } }) => updateWidth(parseInt(value))} />
+      </div>
+      <div className="section-item-content">
+        <div className="section-item-label">Height</div>
+        <Input value={`${height}`} type="number" onChange={({ detail: { value } }) => updateHeight(parseInt(value))} />
+      </div>
+    </SettingTile>
+  );
+};
+
+const PositionSettings = () => {
+  const [x, updateX] = useInput<number>('x');
+  const [y, updateY] = useInput<number>('y');
+  return (
+    <SettingTile title={'Position'} gridDefinition={[{ colspan: 6 }, { colspan: 6 }]}>
+      <div className="section-item-content">
+        <div className="section-item-label">X</div>
+        <Input value={`${x}`} type="number" onChange={({ detail: { value } }) => updateX(parseInt(value))} />
+      </div>
+      <div className="section-item-content">
+        <div className="section-item-label">Y</div>
+        <Input value={`${y}`} type="number" onChange={({ detail: { value } }) => updateY(parseInt(value))} />
+      </div>
+    </SettingTile>
+  );
+};
+
+const YAxisSettings = () => {
+  const [title, updateTitle] = useInput<string>('properties.axis.labels.yAxis.content');
+  const [showAxis = false, updateShowAxis] = useInput<boolean>('properties.axis.showY');
+  return (
+    <SettingTile title={'Y-axis'} gridDefinition={[{ colspan: 12 }, { colspan: 4 }]}>
+      <div className="section-item-content">
+        <div className="section-item-label">Title</div>
+        <Input value={title} onChange={({ detail: { value } }) => updateTitle(value)} />
+      </div>
+      <div className="section-item-content">
+        <div className="section-item-label">Show</div>
+        <Checkbox
+          checked={showAxis}
+          onChange={({ detail: { checked } }) => {
+            updateShowAxis(checked);
+          }}
+        />
+      </div>
+    </SettingTile>
+  );
+};
+const LegendSettings = () => {
+  const [width = 0, updateWidth] = useInput<number>('properties.legend.width');
+  const [position = 'Right', updatePosition] = useInput<LEGEND_POSITION>('properties.legend.position');
+  const [title = '', updateTitle] = useInput<string>('properties.legend.legendLabels.title');
+  return (
+    <SettingTile title="Legend" gridDefinition={[{ colspan: 6 }, { colspan: 6 }, { colspan: 6 }]}>
+      <div className="section-item-content">
+        <div className="section-item-label">Width</div>
+        <Input value={`${width}`} onChange={({ detail: { value } }) => updateWidth(parseInt(value))} />
```

Comment:
Could the lambdas be pulled out into functions? I'm not sure what some are doing and, generally, I find it difficult to read and find anonymous lambdas embedded in JSX.

Reply:
Updated =)